docs/DeveloperGuide.md

```diff
@@ -2,14 +2,191 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
```

Comment:
Perhaps you could include class diagrams for the various components showing the overall interactions and relationships between key components used in your application.